### DISCUSSION TITLE
Fixing a random seed

### DISCUSSION THREAD

**Discussion by AKHCE** [MilesCranmer/PySR#386] [Q&A] [answered]
Hi

As far as I know, initial population and mutation are at least two exemplars where randomness comes into play in symbolic regression. I appreciate you kindly explaining how we can control seed of the random functions in PySr in order to reproducer the results if needed.

Cheers,

**Comment by MilesCranmer** [comment, marked as answer]
See https://astroautomata.com/PySR/api/#performance-and-parallelization

for example
```
random_state=42,
deterministic=True,
procs=0,
multithreading=False,
```
Would get deterministic results with a seed of 42.

Note it’s slower as determinism requires that we can’t use multiple threads.
